### ISSUE TITLE
Seq and SeqRecord integer multiplication

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython#1659]
Python strings support multiplication by integers as follows:

```python
>>> 4 * "A"
'AAAA'
>>> 4 * "ACGT"
'ACGTACGTACGTACGT'
>>> "ACGT" * 4
'ACGTACGTACGTACGT'
```

Note multiplication by floats is not supported:

```
>>> "ACGT" * 4.5
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't multiply sequence by non-int of type 'float'
>>> "ACGT" * 4.0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't multiply sequence by non-int of type 'float'
```

It would be nice to define the relevant methods in the ``Seq`` object and ``UnknownSeq`` object (although it is less clear to me what that ``MutableSeq`` should do here - presumably the same, return a new object rather than acting in place?).

Likewise for the ``SeqRecord`` object, which should probably follow the existing meta-data behaviour for addition. i.e. ```record * 3`` and ```3 * record`` and ``record + record + record`` should all return the same thing.

This would be particularly useful for constructing test cases (e.g. making a ``SeqRecord`` of 1000 base pairs long using `"ACGT" * 25` as the sequence).

**Comment by chris-rands** [comment]
Regarding your question, maybe `MutableSeq` should return a new object and not in in-place in the same way that in native Python `my_list * 3` returns a new list and leaves the original unchanged. This also feels right in terms of methods, I guess `__mul__` for `record * 3`, `__rmul__` for  `3 * record`, and `__imul__` for `record *= 3` (the in-place case).

**Comment by chris-rands** [comment]
Thanks Peter, string literals don't support `*=`, but strings assigned to variables do (as do other immutable types like tuples):

```
>>> seq = "ACGT"
>>> seq *= 3
>>> seq
'ACGTACGTACGT'
```

Also `Seq` objects already support `+=` (without `__iadd__` defined in `Seq.py`, I think python automatically overides it?):

```
>>> from Bio.Seq import Seq
>>> seq = Seq("A")
>>> seq += "T"
>>> seq
Seq('AT', Alphabet())
```

In the Python [docs](https://docs.python.org/3/reference/datamodel.html#emulating-container-types) they say "sequence types should implement addition (meaning concatenation) and multiplication (meaning repetition) by defining the methods \__add__(), \__radd__(), \__iadd__(), \__mul__(), \__rmul__() and \__imul__()",

(but I don't know if `Seq` is meant to be implemented as a sequence type; `isinstance(Seq('A'), collections.Sequence)` is `False`)

**Comment by peterjc** [comment]
Good point - I've never used in-place sequence multiplication so missed that.

I would say ``Seq``, ``UnknownSeq``, ``MutableSeq`` etc are all sequence types, even if not using ``collections.Sequence`` which is comparatively new in Python.

Do you fancy working on this with a pull request?

**Comment by chris-rands** [comment]
I would be happy to work on a pull request, but I'm fairly busy right now, can you wait a few weeks or more for this?

**Comment by peterjc** [comment]
That'd be great. This is not urgent, I filed this issue in part as a public TODO note for myself, but also in case someone else wanted to tackle it.

Note also we'd want to test this with the ``BioSQL`` subclasses - I have some thoughts on that, but in the first instance raising a not implemented exception might be simplest.

**Comment by peterjc** [comment]
Closed via #1669, thanks Chris!
